### ISSUE TITLE
Edit pageSize number

### ISSUE THREAD

**Issue by kamou31** [bencripps/react-redux-grid#175]
Hello,

I'm trying to give users the ability to change the pageSize of the grid.
I created a state variable : itemsDisplayed.
The value is well modified by other part of my code. But unfortunately, even if the render function is called after the update of this state value, the number of items that my grid displays on each page doesn't change. 

Here's my code : 
```
       var itemsNumber = 5;
        if (this.state.itemsDisplayed) {
            itemsNumber = this.state.itemsDisplayed;
        }
        console.log(itemsNumber); // Display 5 then 6 then 7 then 8 after my setState
        const gridData = {
            columns,
            data: data,
            pageSize: itemsNumber,
            plugins: plugins,
            events,
            height: this.state.height,
            stateKey: 'events'
        };
```

The console proves that my itemsNumber is well updated but the grid always display only the first 5 items as it was defined the first time.

Can you help me? Thank you very much !

**Comment by bencripps** [comment]
I would take a look at the bulk selection example, which is dynamically changing what the pageSize. Here is the [code](https://github.com/bencripps/react-redux-grid-demo/tree/master/src/components/examples/bulk-selection), and here is the actual live [example](http://react-redux-grid.herokuapp.com/BulkSelection).

**Comment by kamou31** [comment]
Thank you. I'll try to understand it and I'll keep you in touch.

**Comment by kamou31** [comment]
Thank you, didn't understant the utility of getAsyncData but now it's ok !

Thank you for your help

**Comment by kamou31** [comment]
Indeed, it works, but in the example (like in my code) the sorting parameters aren't kept after the getAsyncData. I'm pretty sure it's doable to keep them with the extraParams but I'm not able to find the current state of my columns sortings. 
Where can i find them? 

Thank you very much

**Comment by GOVYANSONG** [comment]
@bencripps a question: does change page size work with local data? The example is for remote data.
